BuildNotifications.Core/Config/IProjectConfiguration.cs

```diff
@@ -10,28 +10,28 @@ public interface IProjectConfiguration
         /// <summary>
         /// List of branches to never load information for.
         /// </summary>
-        IList<string> BranchBlacklist { get; }
+        IList<string> BranchBlacklist { get; set; }
 
         /// <summary>
         /// List of branches to always load information for.
         /// </summary>
-        IList<string> BranchWhitelist { get; }
+        IList<string> BranchWhitelist { get; set; }
 
         /// <summary>
         /// Names of the <see cref="ConnectionData" /> that are used to
         /// fetch builds for this project.
         /// </summary>
-        IList<string> BuildConnectionNames { get; }
+        IList<string> BuildConnectionName { get; set; }
 
         /// <summary>
         /// List of build definitions to never load information for.
         /// </summary>
-        IList<string> BuildDefinitionBlacklist { get; }
+        IList<string> BuildDefinitionBlacklist { get; set; }
 
         /// <summary>
         /// List of build definitions to always load information for.
         /// </summary>
-        IList<string> BuildDefinitionWhitelist { get; }
+        IList<string> BuildDefinitionWhitelist { get; set; }
```

Comment:
The setter is used by the Configuration ViewModels to persist the settings done by the user. Since there are setters for every other property in the class, I see no benefit in dealing with these lists in a different way

Reply:
Because all other properties are copy-by-value. When you retrieve Configuration.PullRequestDisplay you know that this value is immutable. If Configuration changes, your retrieved PullRequestDisplay value will not change.
With IList this is ambiguous. It's a reference type and it COULD change. With a public setter the IList content could be updated or the IList itself. Without a setter however, it is much more clear that the list instance will update.

An alternative would be to make it a IReadOnlyList, which, vice versa, makes it clear that your instance will not change.

Reply:
Good point. Changed the type to IReadOnlyList.